tests/test_file_listing.py

```
@@ -0,0 +1,123 @@
+        "src/utils.py",
+        "tests/test_main.py",
+        "docs/index.html"
+    ]
+    
+    for file_path in files:
+        full_path = empty_git_repo / file_path
+        full_path.parent.mkdir(parents=True, exist_ok=True)
+        full_path.write_text(f"Content of {file_path}")
+    
+    # Add and commit files
+    subprocess.run(["git", "add", "."], cwd=empty_git_repo)
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)